libraries/ms-common/inc/spi.h

```diff
@@ -0,0 +1,111 @@
+// SPI HAL Interface
+
+#include "interrupt.h"
+#include "status.h"
+#include "stm32f0xx.h"
+
+typdef void (*spi_callback)(SPIPeriph spi_x, void *context);
```

Comment:
If you registered a callback, how would you check which interrupt was triggered?